test/periphery/integration/merkle-campaign/ll/claim/claim.t.sol

```diff
@@ -2,106 +2,23 @@
-        merkleLL.claim({ index: 1, recipient: users.recipient1, amount: 1, merkleProof: merkleProof });
-    }
-
-    modifier givenCampaignNotExpired() {
-        _;
-    }
-
-    function test_RevertGiven_AlreadyClaimed() external givenCampaignNotExpired {
-        claimLL();
-        uint256 index1 = defaults.INDEX1();
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_StreamClaimed.selector, index1));
-        merkleLL.claim(index1, users.recipient1, amount, merkleProof);
-    }
-
-    modifier givenNotClaimed() {
-        _;
-    }
-
-    modifier givenNotIncludedInMerkleTree() {
-        _;
-    }
-
-    function test_RevertWhen_InvalidIndex()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 invalidIndex = 1337;
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleLL.claim(invalidIndex, users.recipient1, amount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidRecipient()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        address invalidRecipient = address(1337);
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleLL.claim(index1, invalidRecipient, amount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidAmount()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        uint128 invalidAmount = 1337;
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleLL.claim(index1, users.recipient1, invalidAmount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidMerkleProof()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory invalidMerkleProof = defaults.index2Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleLL.claim(index1, users.recipient1, amount, invalidMerkleProof);
-    }
-
-    modifier givenIncludedInMerkleTree() {
-        _;
+contract Claim_MerkleLL_Integration_Test is MerkleCampaign_Integration_Shared_Test {
```

Comment:
You have written this test contract here and the same test contract also exists in `MerkleLL.t.sol`. I am not sure why you did that.

```solidity
contract Claim_MerkleLL_Integration_Test is Claim_Integration_Test, MerkleLL_Integration_Shared_Test {
    function setUp() public override(Claim_Integration_Test, MerkleLL_Integration_Shared_Test) {
        Claim_Integration_Test.setUp();
        MerkleLL_Integration_Shared_Test.setUp();
    }
}
```

Since they are technically a duplicate of each other, we should either have them here or in there.

Reply:
the reason was that we would have had two tests with the same name `test_Claim` but i guess we can name it `test_ClaimLL`

Reply:
OK but why is there a need to have two separate test contracts for claim? Can we not merge them into one somehow? The guidelines that we have been using so far seem to be that each function should have one test contract whereas in this case, we have two `Claim` test contracts testing the `claim` function.

From an external reviewer,  if I have seen a Claim test contract in one place, I would not expect that there could be another Claim test contract.

The tree problem that you mentioned in the description also arises from it.